lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -0,0 +1,132 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.JsBuiltIn;
+
+    let FunctionsEnum = {
+        ArrayValues: { className: "Array", methodName: "values", argumentsCount: 0, forceInline: true /*optional*/, alias: "Symbol.iterator" },
+        ArrayKeys: { className: "Array", methodName: "keys", argumentsCount: 0, forceInline: true /*optional*/ },
+        ArrayEntries: { className: "Array", methodName: "entries", argumentsCount: 0, forceInline: true /*optional*/ }
+    };
+
+    var setPrototype = platform.builtInSetPrototype;
+    var _objectDefineProperty = platform.builtInJavascriptObjectEntryDefineProperty;
+
+    // Object's getter and setter can get overriden on the prototype, in that case while setting the value attributes, we will end up with TypeError
+    // So, we need to set the prototype of attributes to null
+    var ObjectDefineProperty = function (obj, prop, attributes) {
+        _objectDefineProperty(obj, prop, setPrototype(attributes, null));
+    };
+    var CreateObject = platform.builtInJavascriptObjectCreate;
+
+    platform.registerChakraLibraryFunction("ArrayIterator", function (arrayObj, iterationKind) {
+        "use strict";
+        __chakraLibrary.InitInternalProperties(this, 4, "__$arrayObj$__", "__$nextIndex$__", "__$kind$__", "__$internalDone$__");
+        this.__$arrayObj$__ = arrayObj;
```

Comment:
>this.__$arrayObj$__ [](start = 8, length = 19)

make sure you speak with Louis/Ed about this.

Reply:
Done. Got approval.